src/coreclr/src/vm/dllimportcallback.h

```diff
@@ -198,31 +184,33 @@ class UMThunkMarshInfo
         return (UINT32)offsetof(UMThunkMarshInfo, m_pILStub);
     }
 
-#if defined(TARGET_X86) && !defined(FEATURE_STUBS_AS_IL)
-    // Compiles an unmanaged to managed thunk for the given signature. The thunk
-    // will call the stub or, if fNoStub == TRUE, directly the managed target.
-    Stub *CompileNExportThunk(LoaderHeap *pLoaderHeap, PInvokeStaticSigInfo* pSigInfo, MetaSig *pMetaSig, BOOL fNoStub);
-#endif // TARGET_X86 && !FEATURE_STUBS_AS_IL
+#ifdef TARGET_X86
 
-#if defined(TARGET_X86) && defined(FEATURE_STUBS_AS_IL)
+#ifdef FEATURE_STUBS_AS_IL
     struct ArgumentRegisters
     {
         UINT32 Ecx;
         UINT32 Edx;
     };
 
     VOID SetupArguments(char *pSrc, ArgumentRegisters *pArgRegs, char *pDst);
```

Comment:
This argument shuffling needs to be fixed to make the nativecallable path usable for non-Windows x86.